service/scaladsl/kafka/server/src/main/scala/com/lightbend/lagom/scaladsl/broker/kafka/LagomKafkaComponents.scala

```diff
@@ -16,6 +16,12 @@ trait LagomKafkaComponents extends LagomKafkaClientComponents {
   def lagomServer: LagomServer
   def offsetStore: OffsetStore
 
+  override def topicPublisherName: Option[String] = super.topicPublisherName match {
+    case Some(other) =>
+      sys.error(s"Cannot provide the kafka topic factory as the default topic publisher since a default topic publisher has already been mixed into this cake: $other")
```

Comment:
Will everyone understand what "mixed into this cake" means?

Maybe it could say something like "since a default topic provider has already been provided by a superclass"?